apiserver/facades/client/spaces/opfactory.go

```diff
@@ -48,7 +52,13 @@ func (f *opFactory) NewRenameSpaceOp(fromName, toName string) (state.ModelOperat
 		return nil, errors.Trace(err)
 	}
 	return NewRenameSpaceOp(
```

Comment:
@manadart I had thought of something different (keeping the operation because it updates the constraints and the settings) but naively rollbacking if the dqlite rename fails, this is ongoing on https://github.com/juju/juju/pull/17126/commits/47881f87bed79c62056ec2149ee673a8a878a534